fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -122,6 +138,41 @@ private void preProcessConjuncts() {
         icebergPredicates = expressions;
     }
 
+    /**
+     * Append columns need to read for equality delete files
+     */
+    public void appendEqualityColumns(PhysicalIcebergScanOperator node,
+                                      ColumnRefFactory columnRefFactory,
+                                      ExecPlan context) {
+        Table referenceTable = node.getTable();
+        Set<String> scanNodeColumns = node.getColRefToColumnMetaMap().values().stream()
+                .map(column -> column.getName()).collect(Collectors.toSet());
+        Set<String> appendEqualityColumns = equalityDeleteColumns.stream()
+                .filter(name -> !scanNodeColumns.contains(name)).collect(Collectors.toSet());
+        Map<String, Object> nameToColumns = referenceTable.getFullSchema().stream().collect(Collectors.toMap(Column::getName, item -> item));
```

Comment:
nit:
1. There are over 130 in this line. 
2. Map<String, Object> -> Map<String, Column>